patches/tModLoader/Terraria/ModLoader/TileLoader.cs

```diff
@@ -485,29 +495,74 @@ public static void DropCritterChance(int i, int j, int type, ref int wormChance,
 		}
 	}
 
+	// TODO: Drop is usually called after vanilla multitiles are already killed, meaning frame information is missing. 
+	// i and j are the coordinates being checked, not the top left. (Should it be top left?)
+	// Could potentially change this to be called in Item.NewItem when EntitySource_TileBreak, but that is used for TEs dropping container items as well. Also tile type is unrecoverable at that point.
 	public static bool Drop(int i, int j, int type)
 	{
-		foreach (var hook in HookDrop) {
+		bool dropItem = true;
+		ModTile modTile = GetTile(type);
+		foreach (var hook in HookCanDrop) {
 			if (!hook(i, j, type)) {
-				return false;
+				dropItem = false;
 			}
 		}
+		if (modTile != null) {
+			dropItem &= modTile.CanDrop(i, j);
+		}
```

Comment:
Suggest
`bool dropItem = modTile?.CanDrop(i, j) ?? true;`
and
`dropItem &= hook(i, j, type)`